diff --git a/src/libevp-agent/meson.build b/src/libevp-agent/meson.build
index 9c1facc..d3e7e53 100644
--- a/src/libevp-agent/meson.build
+++ b/src/libevp-agent/meson.build
@@ -194,6 +194,7 @@ evp_agent = static_library(
 		mbedx509_dep,
 		parson_dep,
 		wamr_dep,
+		sdkenc_dep,
 	]
 )
 
diff --git a/src/sdkenc/meson.build b/src/sdkenc/meson.build
index d31f678..b697eb0 100644
--- a/src/sdkenc/meson.build
+++ b/src/sdkenc/meson.build
@@ -31,3 +31,7 @@ else
 		build_by_default : true,
 	)
 endif
+
+sdkenc_dep = declare_dependency(
+	sources : sdkenc_headers
+)
