diff --git a/CMakeLists.txt b/CMakeLists.txt
index 18bfe08074..9edae60cd6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -107,6 +107,9 @@ set(CTR_DRBG_128_BIT_KEY_WARNING "${WARNING_BORDER}"
                          "${CTR_DRBG_128_BIT_KEY_WARN_L3}"
                          "${WARNING_BORDER}")
 
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+#set(CMAKE_FIND_DEBUG_MODE 1)
+
 # Python 3 is only needed here to check for configuration warnings.
 if(NOT CMAKE_VERSION VERSION_LESS 3.15.0)
     set(Python3_FIND_STRATEGY LOCATION)
@@ -147,6 +150,13 @@ endif()
 set(MBEDTLS_CONFIG_FILE "" CACHE FILEPATH "Mbed TLS config file (overrides default).")
 set(MBEDTLS_USER_CONFIG_FILE "" CACHE FILEPATH "Mbed TLS user config file (appended to default).")
 
+set(USER_CONFIG_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/../packagefiles/mbedtls/include/mbedtls/user_config.h")
+if(EXISTS "${USER_CONFIG_SOURCES}")
+    file(COPY "${USER_CONFIG_SOURCES}" DESTINATION "${CMAKE_CURRENT_SOURCE_DIR}/include/mbedtls/")
+else()
+    message(WARNING "User config file not found: ${USER_CONFIG_SOURCES}")
+endif()
+
 # Create a symbolic link from ${base_name} in the binary directory
 # to the corresponding path in the source directory.
 # Note: Copies the file(s) on Windows.
